app/code/Magento/Sales/Block/Order/Totals.php

```diff
@@ -126,16 +126,21 @@ protected function _initTotals()
          * Add discount
          */
         if ((double)$this->getSource()->getDiscountAmount() != 0) {
-            if ($this->getSource()->getDiscountDescription()) {
-                $discountLabel = __('Discount (%1)', $source->getDiscountDescription());
+            if ($this->getSource() instanceof Invoice) {
+                $discountSource = $this->getSource()->getOrder();
+            } else {
+                $discountSource = $this->getSource();
+            }
+            if ($this->getSource()->getCouponCode()) {
+                $discountLabel = __('Discount (%1)', $discountSource->getCouponCode());
```

Comment:
As I see from the suggested solution, the coupon code description will not be shown anymore, but it was expected to be shown. In other words, this solution replaces coupon code description with its code. It is not correct from the user experience.
It is also impossible to change the `$discountLabel` text because this class is an `@api` (BIC change).

I can suggest to implement something like this:
```php
            $discountLabel = $this->getSource()->getDiscountDescription()
                ? __('Discount (%1)', $source->getDiscountDescription())
                : __('Discount (%1)', $source->getCouponCode());
```
With this, we will have a coupon code or description in the Discount line. And not the BIC change.